massa-execution-worker/src/execution.rs

```diff
@@ -1290,36 +1319,23 @@ impl ExecutionState {
         // check if the final slot execution result is already cached at the front of the speculative execution history
         let first_exec_output = self.active_history.write().0.pop_front();
         if let Some(exec_out) = first_exec_output {
-            if &exec_out.slot == slot && exec_out.block_id == target_id {
+            if &exec_out.slot == slot
+                && exec_out.block_info.as_ref().map(|i| i.block_id) == target_id
+            {
                 // speculative execution front result matches what we want to compute
 
+                // update versioning stats
+                // Note: This update the MIP Store and this must be done before apply_final_execution_output
+                //       as it will write the final state (and thus the MIP store) to the DB
+                self.update_versioning_stats(&exec_out.block_info, slot);
```

Comment:
could it just be done at the beginning of apply_final_execution_output to avoid duplication ?

Reply:
done